veps/sig-network/ns-admin-nichotplug.md

```diff
@@ -0,0 +1,225 @@
+```
+
+Existing hotplug / hotunplug flows could be adjusted to use it.
+
+#### Migration Based Hotplug Flow
+
+1. The VM owner adds one or more new secondary vNICs and matching networks to a running VM.
+2. The VM controller patches the VMI object with the additional relevant vNICs and networks.
+3. The VMI controller:
+   
+   - Patches the virt-launcher pod's Multus network attachment annotation (`k8s.v1.cni.cncf.io/networks`) - since the dynamic-networks-controller is not present, this will not have an affect.
+   - Updates the VMI interface status.
+   - Adds the `MigrationRequired` condition to the VMI with its status set to `False` and a timestamp.
+4. On a future iteration of the reconciliation loop, the VMI controller will update the `MigrationRequired` condition status to `True` (Following no change in the interface statuses for more than TBD seconds after the initial timestamp set on the condition).
+5. The Workload update controller will identify the VMI as a candidate for automatic migration and will create a VMIM object for it.
+6. The Migration controller will remove the `MigrationRequired` condition once the migration is complete.
+7. The vNICs will be hot-plugged following the migration - same as today. 
+
+#### Migration Based Hotunplug Flow
+
+1. The VM owner sets the state one or more existing secondary vNICs to `absent`.
+2. The VM controller patches the VMI object with the relevant interfaces and networks.
+3. The VMI controller:
+
+    - Patches the virt-launcher pod's Multus network attachment annotation (`k8s.v1.cni.cncf.io/networks`) - since the dynamic-networks-controller is not present, this will not have an affect.
+    - Updates the VMI interface status.
+    - Adds the `MigrationRequired` condition to the VMI with its status set to `False` and a timestamp.
+4. On a future iteration of the reconciliation loop, the VMI controller will update the `MigrationRequired` condition status to `True` (Following no change in the interface statuses for more than TBD seconds after the initial timestamp set on the condition).
+5. The Workload update controller will identify the VMI as a candidate for automatic migration and will create a VMIM object for it.
+6. The Migration controller will remove the `MigrationRequired` condition once the migration is complete.
+7. The vNICs will be hot-unplugged following the migration - same as today.
+
+#### In-Place Hotplug / Hotunplug Flow
+
+1. The VM owner adds one or more new secondary vNICs with bridge binding and matching networks, or sets their state to `absent` to / on a running VM.
+2. The VM controller patches the VMI object with the additional relevant vNICs and networks.
+3. The VMI controller:
+
+    - Patches the virt-launcher pod's Multus network attachment annotation (`k8s.v1.cni.cncf.io/networks`).
+    - Updates the VMI interface status.
+    - Adds the `MigrationRequired` condition to the VMI with the `False` with its status set to `False` and a timestamp.
```

Comment:
Something went wrong with the sentence here (double `False`).

Reply:
Thanks.
Done.